From 2a3710872f8bfc1c95dc8044c92c9ce020adad40 Mon Sep 17 00:00:00 2001
From: Alexander Larsson <alexl@redhat.com>
Date: Thu, 08 Oct 2009 09:59:23 +0000
Subject: xdgmime: sort glob hits with larger weight first

Higher weight is more important, so return these first meaning
they will be used as the default.
---
diff --git a/gio/xdgmime/xdgmimecache.c b/gio/xdgmime/xdgmimecache.c
index 6765787..97fe975 100644
--- a/gio/xdgmime/xdgmimecache.c
+++ b/gio/xdgmime/xdgmimecache.c
@@ -565,7 +565,7 @@ static int compare_mime_weight (const void *a, const void *b)
   const MimeWeight *aa = (const MimeWeight *)a;
   const MimeWeight *bb = (const MimeWeight *)b;
 
-  return aa->weight - bb->weight;
+  return bb->weight - aa->weight;
 }
 
 #define ISUPPER(c)		((c) >= 'A' && (c) <= 'Z')
diff --git a/gio/xdgmime/xdgmimeglob.c b/gio/xdgmime/xdgmimeglob.c
index 2a0c777..c82bf22 100644
--- a/gio/xdgmime/xdgmimeglob.c
+++ b/gio/xdgmime/xdgmimeglob.c
@@ -381,7 +381,7 @@ static int compare_mime_weight (const void *a, const void *b)
   const MimeWeight *aa = (const MimeWeight *)a;
   const MimeWeight *bb = (const MimeWeight *)b;
 
-  return aa->weight - bb->weight;
+  return bb->weight - aa->weight;
 }
 
 #define ISUPPER(c)		((c) >= 'A' && (c) <= 'Z')
--
cgit v0.8.2
